docs/END-TO-END-TESTING.md

```diff
@@ -0,0 +1,103 @@
+
+Follow the links below one-by-one to retrieve required information from Google:
+1. [Google Project and Application Setup](https://docs.cypress.io/guides/testing-strategies/google-authentication#Google-Developer-Console-Setup)
+2. [Using the Google OAuth 2.0 Playground to Create Testing Credentials](https://docs.cypress.io/guides/testing-strategies/google-authentication#Using-the-Google-OAuth-2-0-Playground-to-Create-Testing-Credentials)
+
+### Installation
+1. Install TestCafe for BrowserStack globally:
+    ```sh
+    npm i -g testcafe testcafe-browser-provider-browserstack
+    ```
+
+2. Add the following env variables to your `~/.bash_profile` or `~/.zshrc` file:
+    ```sh
+    export BROWSERSTACK_BASE_URL="http://localhost:8080"
+    export BROWSERSTACK_USERNAME="" # Your BrowserStack username
+    export BROWSERSTACK_ACCESS_KEY="" # Your BrowserStack access key
+    export BROWSERSTACK_PROJECT_NAME="plio"
+    ```
+
+3. Once the env variables are added, restart your terminal or run one of the following command:
+    ```sh
+    source ~/.bash_profile
+    # or
+    source ~/.zshrc
+    ```
+
+### Getting started
+1. To run a test case locally, execute the following command from the project root folder:
+    ```sh
+    # syntax
+    testcafe browser ./path/to/test.spec.js
+
+    # example
+    testcafe chrome ./tests/integration/pages/login.spec.js
+    ```
+2. To list all the browsers available, run the following command:
+    ```sh
+    # command to see local browsers
+    testcafe --list-browsers
+
+    # command to see BrowserStack browsers
+    testcafe -b browserstack
+    ```
+
+
+### Environment variables
+Setup the following environment variables in your `~/.bash_profile` or `~/.zshrc` file if you're trying to run testcafe locally with BrowserStack. Also, set them up on the Continuous Integration environment using GitHub secrets.
+
+#### `BROWSERSTACK_BASE_URL`
+The base url of the application registered with BrowserStack. Should be the frontend application URL.
+
+#### `BROWSERSTACK_USERNAME`
+Required to connect to your BrowserStack account. Get this information from your BrowserStack account settings.
+
+#### `BROWSERSTACK_ACCESS_KEY`
+Required to connect to your BrowserStack account. Get this information from your BrowserStack account settings.
+
+#### `BROWSERSTACK_PROJECT_NAME`
+Name of the BrowserStack project. Set to `plio`. It allows to name the build, which is useful for keeping track of test runs on the BrowserStack dashboard.
+
+### Environment variables (only for Continuous Integration)
+The following environment variables are only required for the Continuous Integration environment. These are used to install and build the frontend and backend application. Set these up using GitHub secrets only and not in the `.env` files.
+
+#### `GOOGLE_OAUTH2_CLIENT_ID`
```

Comment:
why not re-use `VUE_APP_GOOGLE_CLIENT_ID`? and keep other variable names as `VUE_APP_*` too?